modules/localgov_review_date/tests/src/FunctionalJavascript/ReviewDateWidgetTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\Tests\localgov_review_date\FunctionalJavascript;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\localgov_review_date\Form\ReviewDateSettingsForm;
+use Drupal\workflows\Entity\Workflow;
+
+/**
+ * Tests the Review Status widget UI.
+ *
+ * @group path
+ */
+class ReviewDateWidgetTest extends WebDriverTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $profile = 'testing';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'claro';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'localgov_review_date',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Create a page content type.
+    $this->drupalCreateContentType(['type' => 'page', 'name' => 'Basic page']);
+
+    // Add page to localgov_editorial workflow.
+    $editorial = Workflow::load('localgov_editorial');
+    $type = $editorial->getTypePlugin();
+    $type->addEntityTypeAndBundle('node', 'page');
+    $editorial->save();
+
+    // Configure scheduled transitions.
+    $scheduled_transitions_config = \Drupal::service('config.factory')->getEditable('scheduled_transitions.settings');
+    $bundles = [
+      [
+        'entity_type' => 'node',
+        'bundle' => 'page',
+      ],
+    ];
+    $scheduled_transitions_config->set('bundles', $bundles);
+    $scheduled_transitions_config->save();
+
+    $this->drupalLogin($this->rootUser);
+  }
+
+  /**
+   * Test review date widget actions.
+   */
+  public function testReviewDateWidget() {
+
+    // Check initial settings.
+    $this->drupalGet('node/add/page');
+    $page = $this->getSession()->getPage();
+    $page->hasUncheckedField('localgov_review_date[0][reviewed]');
+    $container = $page->find('css', '.review-date-container');
+    $this->assertNotTrue($container->isVisible());
+
+    // Check display when once checked.
+    $page->checkField('localgov_review_date[0][reviewed]');
+    $this->assertTrue($container->isVisible());
+
+    // Check next review date selector.
+    foreach (ReviewDateSettingsForm::getNextReviewOptions() as $month => $description) {
+      $review_in = date('Y-m-d', strtotime('+' . $month . ' months'));
```

Comment:
I fear this test, probably the test rather than the code it's testing, might be prone to underlying timezone failures. 

I'm presently getting:

--- Expected
+++ Actual
@@ @@
-'2022-01-02'
+'2022-01-01'

I'll investigate further. IIRC Drupal tests, or at least core, was running with a timezone of Australia/Sydney

Reply:
Yep code isn't happy in the early Australian morning I'd say 
https://github.com/localgovdrupal/localgov_workflows/pull/18/checks?check_run_id=3769485313

Reply:
Updated to use a set timestamp rather than now

Reply:
I have set the timezone in Drupal to try and fix this issue. The tests pass now, but I don't know if this us just because they didn't hit the timezone issue. Not sure what else can be done here beyond removing this test.

https://github.com/localgovdrupal/localgov_workflows/blob/e683fae983afa696cf3d08055432bde4bb529d5e/modules/localgov_review_date/tests/src/FunctionalJavascript/ReviewDateWidgetTest.php#L73-L75